python/ray/tune/examples/pbt_transformers.py

```diff
@@ -0,0 +1,225 @@
+    )
+    eval_dataset = (
+        GlueDataset(data_args, tokenizer=tokenizer, mode="dev", cache_dir=model_args.cache_dir)
+        if training_args.do_eval
+        else None
+    )
+
+    def build_compute_metrics_fn(task_name: str) -> Callable[[EvalPrediction], Dict]:
+        def compute_metrics_fn(p: EvalPrediction):
+            if output_mode == "classification":
+                preds = np.argmax(p.predictions, axis=1)
+            elif output_mode == "regression":
+                preds = np.squeeze(p.predictions)
+            metrics = glue_compute_metrics(task_name, preds, p.label_ids)
+            tune.report(mean_accuracy=metrics["acc"])
+            return metrics
+
+        return compute_metrics_fn
+
+    trainer = Trainer(
+        model=model,
+        args=training_args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        compute_metrics=build_compute_metrics_fn(data_args.task_name)
+    )
+    return trainer
+# __huggingface_end__
+
+
+# __download_begin__
+def download_data(model_name, task_name, data_dir="./data"):
+    # Triggers tokenizer download to cache
+    AutoTokenizer.from_pretrained(model_name)
+
+    # Triggers model download to cache
+    AutoModelForSequenceClassification.from_pretrained(
+        model_name,
+        from_tf=bool(".ckpt" in model_name)
+    )
+
+    # Download RTE training data
+    import urllib
+    import zipfile
+    if task_name == "rte":
+        url = "https://firebasestorage.googleapis.com/v0/b/mtl-sentence-representations.appspot.com/o/data%2FRTE.zip?alt=media&token=5efa7e85-a0bb-4f19-8ea2-9e1840f077fb"
+    else:
+        raise ValueError("Unknown task: {}".format(task_name))
+    data_file = os.path.join(data_dir, "{}.zip".format(task_name))
+    if not os.path.exists(data_file):
+        urllib.request.urlretrieve(url, data_file)
+        with zipfile.ZipFile(data_file) as zip_ref:
+            zip_ref.extractall(data_dir)
+        print("Downloaded data for task {} to {}".format(task_name, data_dir))
+# __download_end__
+
+
+# __train_begin__
+def train_transformer(config, model_name, task_name, num_epochs=3, data_dir="./data"):
+    data_args = DataTrainingArguments(
+        task_name=task_name,
+        data_dir=data_dir
+    )
+    model_args = ModelArguments(
+        model_name_or_path=model_name
+    )
+    training_args = TrainingArguments(
+        output_dir=tune.get_trial_dir(),
+        learning_rate=config["lr"],
+        do_train=True,
+        do_eval=True,
+        evaluate_during_training=True,
+        eval_steps=10,
+        num_train_epochs=num_epochs,
+        per_device_train_batch_size=config["batch_size"],
+        per_device_eval_batch_size=config["batch_size"],
+        warmup_steps=0,
+        weight_decay=config["weight_decay"],
+        logging_dir="./logs",
+    )
+
+    trainer = get_trainer(data_args, model_args, training_args)
+    trainer.train(model_name)
```

Comment:
btw, we need to capture the checkpoint somehow in order to do pbt. I don't see a _great_ way other than to subclass the Trainer and override its save_model class. 

Thoughts?